indexer/src/transform.test.ts

```diff
@@ -0,0 +1,606 @@
+        "0x4d6aa6a7",
+        "0x7f",
+        "0x0",
+      ],
+      data: [],
+      index: 3,
+    },
+    {
+      fromAddress:
+        "0x4f57d7d15274d9fe832c88aafd80ecc90a8db5696913ddc39cf24af4eba3538",
+      keys: [
+        "0x5ad857f66a5b55f1301ff1ed7e098ac6d4433148f0b72ebc4a2945ab85ad53",
+      ],
+      data: [
+        "0x20",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x0",
+        "0x7f",
+        "0x1",
+        "0x13842",
+      ],
+      index: 4,
+    },
+    {
+      fromAddress:
+        "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
+      keys: [
+        "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9",
+      ],
+      data: [
+        "0x4f57d7d15274d9fe832c88aafd80ecc90a8db5696913ddc39cf24af4eba3538",
+        "0x46a89ae102987331d369645031b49c27738ed096f2789c24449966da4c6de6b",
+        "0x19452efbea3d",
+        "0x0",
+      ],
+      index: 5,
+    },
+  ],
+};
+const event: Event = {
+  fromAddress:
+    "0x4f57d7d15274d9fe832c88aafd80ecc90a8db5696913ddc39cf24af4eba3538",
+  keys: [
+    "0x5ad857f66a5b55f1301ff1ed7e098ac6d4433148f0b72ebc4a2945ab85ad53",
+  ],
+  data: [
+    "0x20",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x7f",
+    "0x1",
+    "0x13842",
+  ],
+  index: 4,
+};
+const events: EventWithTransaction[] = [
+  {
+    transaction,
+    receipt,
+    event,
+  },
+];
+const transactions: TransactionWithReceipt[] = [
+  {
+    transaction,
+    receipt,
+  },
+];
+
+Deno.test("transform with no events or transactions", async () => {
+  const events: EventWithTransaction[] = [];
+  const transactions: TransactionWithReceipt[] = [];
+
+  const result = await transform({ header, events, transactions });
+
+  const expectedHeader = await toEthHeader({
+    header: header,
+    gasUsed: BigInt(0),
+    logsBloom: new Bloom(new Uint8Array(256)),
+    receiptRoot: new Uint8Array(),
+    transactionRoot: new Uint8Array(),
+    blockNumber: padString(toHexString(header.blockNumber), 8),
+    blockHash: header.blockHash,
+    isPendingBlock: false,
+  });
+
+  assertEquals(result.length, 1);
+  assertEquals(result[0].collection, "headers");
+  if ("header" in result[0].data) {
```

Comment:
Here maybe to avoid a big group in assertion we could do something like

```ts
if (!header) throw new Error("Expected header not found in result data");
```

wdyt?

Reply:
I removed all the if-else blocks from the asserts and replaced them with assertHasHeader/receipt/log. what do you think?